cryostatio/cryostat-operator
# test(scorecard): add scorecard test for operator availability

This adds the first custom scorecard test for the operator. This test is pretty basic, it just verifies that the operator deployment becomes available. It should help to serve as a simple smoke test to ensure that the operator can at least start.

I followed the instructions in the Operator SDK guide on custom scorecard tests for the most part. [1] The sample test image for Tekton was also a useful resource [2]. The tests run in a separate namespace with the operator bundle installed and RBAC objects used by the test pod created.

To test it out:
1. Build the custom scorecard test image and push to Quay.
    ```
    export CUSTOM_SCORECARD_IMG="quay.io/<your name>/cryostat-operator-scorecard:some-unique-tag"
    make scorecard-build 
    podman push "$CUSTOM_SCORECARD_IMG"
    ```
2. Generate bundle and push to Quay (with CUSTOM_SCORECARD_IMG defined from step 1).
    ```
    export BUNDLE_IMG="quay.io/<your name>/cryostat-operator-bundle:latest"
    make bundle bundle-build
    podman push "$BUNDLE_IMG"
    ```
3. Run scorecard tests (with running cluster)
    ```
    make test-scorecard
    ```

You should see a result for the custom test like the following:
```
--------------------------------------------------------------------------------
Image:      quay.io/<your name>/cryostat-operator-scorecard:some-unique-tag
Entrypoint: [cryostat-scorecard-tests operator-install]
Labels:
	"test":"operator-install"
	"suite":"cryostat"
Results:
	Name: operator-install
	State: pass

	Log:
		deployment cryostat-operator-controller-manager is available


--------------------------------------------------------------------------------
```

Fixes: #444 

[1] https://sdk.operatorframework.io/docs/testing-operators/scorecard/custom-tests/
[2] https://github.com/operator-framework/tekton-scorecard-image
